lib/controllers/coinstac/consortia.js

```diff
@@ -17,31 +27,143 @@ internals.consortiaSchema = joi.object().keys({
-            path: defaults.path,
-            name: name
-        };
-    } else if (consortiaDbOptions.conn) {
-        pouchWConfig = {
-            name: name,
-            conn: defaults.conn
+    if (isCloudant) {
+        return cloudantClient.mergeSecurity(db.name, security);
+    } else {
+        return Bluebird.resolve();
+    }
+};
+
+/**
+ * compute average of values
+ * @param  {array} values array of objects with `value` and `weight` properties
+ * @return {object}       object with a `value` and `weight` property
+ */
+internals.average = (values) => {
+    return _.reduce(values, (res, value) => {
+        const weightedResVal = (res.value * res.weight);
+        const weightedVal = (value.value * value.weight);
+        const numerator = weightedResVal + weightedVal;
+        const denominator = res.weight + value.weight;
+        res.value = numerator / denominator;
+        res.weight = denominator;
+        return res;
+    });
+};
+
+/**
+ * Add listeners to newly created consortiumDb
+ * @param  {object} db     pouchDB or pouchWrapper insance
+ * @param  {server} server server instance
+ * @return {null}           nothing
+ */
+internals.watchConsortiumDb = (db, server) => {
+    const getAnalyses = (docs) => {
+        return _.filter(docs, (val) => {
+            return !val.aggregate;
+        });
+    };
+
+    const getAggregate = (docs) => {
+        return _.find(docs, {aggregate: true}) || {};
+    };
+
+    const calculateAverage = (values) => {
+        const sum = (arr) => {
+            return _.reduce(arr, (res, val) => {
+                return res + val;
+            }, 0);
         };
+
+        return sum(values) / values.length;
+    };
+
+    const recalcAggregate = (docs) => {
+        const aggregate = getAggregate(docs);
+        const analyses = getAnalyses(docs);
+        aggregate.sampleSize = analyses.length;
+        aggregate.aggregate = true;
+        if (analyses.length) {
+            aggregate.result = calculateAverage(_.map(analyses, 'result'));
+        }
+
+        return aggregate;
+    };
+
+    db.changes({
+        live: true,
+        since: 'now',
+        /*jscs:disable*/
+        include_docs: true //jshint ignore:line
+        /*jscs:enable*/
+    }).on('change', (info) => {
+        if (!info.doc.aggregate) {
+            db.all()
+                .then(recalcAggregate)
+                .then(_.bind(db.save, db))
+                .catch((err) => {
+                    server.log(
+                        ['error', 'aggregate-anlysis', 'coinstac'],
+                        err
+                    );
+                });
+        }
+    });
+};
+
+/**
+ * get a consortia PouchDB client from existing store or by creating one
+ * also calls setConsortiaDbSecurity
+ * @param  {string} name   the consortia id
+ * @param  {function} PouchW the constructor for DB creation
+ * @return {Promise}        resolves to the newly created DB
+ */
+internals.getConsortiumDb = (name, server) => {
```

Comment:
Changed the name from `createConsortiumDb` to `getConsortiumDb`, since it may just retrieve the DB from internals.